lib/OpenQA/WebAPI/Controller/API/V1/Search.pm

```diff
@@ -118,6 +118,25 @@ sub _search_perl_modules {
     return \@results;
 }
 
+sub _search_job_modules {
+    my ($self, $keywords, $limit) = @_;
```

Comment:
How about a sane default that you return in a hash together with @results, e.g. `return {limit => $limit, results => \@results};` and when displaying read out $limit for a message like "Showing the first $limit matches".

Reply:
We already have a configurable value for the limit. This is the remainder after previous results. The changed limit is taken into account after calling this function.

If you're thinking of introducing per-type limits I'd suggest to file a separate ticket, since this PR is just adding job module results.

Reply:
Regrading return values in general: It would likely be more efficient to pass `@results` from the outer scope by reference instead of creating intermediate `@results` arrays in each function. That would also avoid the additional `push` calls in the outer scope. It would also avoid the accounting for the limit in the outer scope (just pass e.g. `{rows => ($limit - scalar @$results)}` to DBIx without ever having to alter `$limit` itself.

Reply:
if the limit is already handled from outside in this way then ok, no need.

@Martchus if it is feasible and easy to finish this week, yes :)

Reply:
> Regrading return values in general: It would likely be more efficient to pass `@results` from the outer scope by reference instead of creating intermediate `@results` arrays in each function. That would also avoid the additional `push` calls in the outer scope. It would also avoid the accounting for the limit in the outer scope (just pass e.g. `{rows => ($limit - scalar @$results)}` to DBIx without ever having to alter `$limit` itself.

That rows trick doesn't work, see my second commit. We need to handle it explicitly either way.